Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,579 @@
+
+    /**
+     *
+     * @param target
+     * @param isRequesting
+     * @param webContextProprties
+     * @param serviceName
+     * @return
+     */
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target, boolean isRequesting, Properties webContextProprties, String serviceName) {
+        String strUrl = null;
+        String strHost = null;
+        boolean setDefaultValue = true;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        strUrl = isRequesting ? getWebServiceUrlFromRemoteObject(target, serviceName) : getWebServiceRequestURL(webContextProprties);
+        if (strUrl != null) {
+            try {
+                URL url = new URL(strUrl);
+                participant.setUserID(strUrl);
+                strHost = url.getHost();
+                participant.setNetworkAccessPointID(strHost);
+                participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(strHost));
+                setDefaultValue = false;
+            } catch (MalformedURLException ex) {
+                LOG.error(ex);
+                setDefaultValue = true;
+            }
+        }
+        //if the url is null or not a valid url
+        if (setDefaultValue) {
+            //for now set the user id to anonymouns
+            participant.setUserID(AuditTransformConstants.ACTIVE_PARTICPANT_USER_ID_SOURCE);
+            //for now hardcode the value to localhost, need to find out if this needs to be set
+            participant.setNetworkAccessPointTypeCode(AuditTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
+            participant.setNetworkAccessPointID(AuditTransformConstants.ACTIVE_PARTICPANT_UNKNOWN_IP_ADDRESS);
+        }
+        participant.setUserIsRequestor(Boolean.FALSE);
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(AuditTransformConstants.ACTIVE_PARTICIPANT_ROLE_CODE_DEST, null,
+            AuditTransformConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME, AuditTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_DESTINATION_DISPLAY_NAME));
+        return participant;
+    }
+
+    /**
+     *
+     * @param webContextProeprties
+     * @return
+     */
+    protected String getRemoteHostAddress(Properties webContextProeprties) {
+        if (webContextProeprties != null && !webContextProeprties.isEmpty() && webContextProeprties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS) != null) {
+            return webContextProeprties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+        }
+        return AuditTransformConstants.ACTIVE_PARTICPANT_UNKNOWN_IP_ADDRESS;
+    }
+
+    /**
+     *
+     * @param hostAddress
+     * @return
+     */
+    protected Short getNetworkAccessPointTypeCode(String hostAddress) {
+        if (InetAddressValidator.getInstance().isValid(hostAddress)) {
+            return AuditTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_IP;
+        } else {
```

Comment:
This is the only place you put an "else" after the "if" returns -- for consistency, change this to not use an else statement.